### PR TITLE
[XPU] DeepSeek V4: use sgl-kernel-xpu implemetation of flash_mla_sparse_fwd for prefill

### PR DESCRIPTION
**Pull request by Valentine233**
Depend on sgl-kernel-xpu implementation https://github.com/sgl-project/sgl-kernel-xpu/pull/337.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #30967052841](https://github.com/sgl-project/sglang/actions/runs/30967052841)
Latest PR Test (Extra): :x: [Run #30967052749](https://github.com/sgl-project/sglang/actions/runs/30967052749)

### CODE REVIEW

File: python/sglang/srt/layers/attention/deepseek_v4_backend.py [hunk 1]
```diff
@@ -1850,6 +1853,9 @@ def _forward_prefill_sparse(
         )
         kv = workspace
 
+        # The XPU op only returns (out, max_logits, lse) when
+        # ``return_softmax_lse`` is set; CUDA returns the 3-tuple by default.
+        extra_kwargs = {"return_softmax_lse": True} if _is_xpu else {}
```
Comment (polisettyvarma) [inline python/sglang/srt/layers/attention/deepseek_v4_backend.py]: XPU can also always return 3 with unused as None return_softmax_lse is False

**Review by polisettyvarma — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/attention/deepseek_v4_backend.py (hunk 1 above)
Comment (Valentine233) [reply python/sglang/srt/layers/attention/deepseek_v4_backend.py]: Thanks. I'll remove the change and update the xpu's kernel.

**Review by airMeng — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]

**Review by polisettyvarma — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]

**Comment by mingfeima** [comment]
rerun failed base CIs. the failures should be flaky. anyway check again.
